macros/parserRadioButtons.pl

```diff
@@ -539,12 +539,11 @@ sub BUTTONS {
     my $value = "B$i"; my $tag = $choices[$i];
     $value = "%".$value if $i == $self->{checkedI};
     $tag = $self->labelFormat($self->{labels}[$i]).$tag if $self->{displayLabels};
-    if ($extend) {
-      push(@radio,main::NAMED_ANS_RADIO_EXTENSION($name,$value,$tag,
-	   aria_label=>$label."option $i "));
+    if ($i > 0) {
```

Comment:
The change from `$extend` to `$i > 0` is incorrect.  The initial `$extend` value comes from an argument to the `BUTTONS()` function, and it is not always 0.  The `named_ans_rule_extension()` method sets it to 1 initially, so `$extend` is tru even when `$i = 0` in that case.   Your modification would change how that is handled.  Note that the `named_ans_rule_extension()` method is used by the MultiAnswer object, so this is an actual use case.

Reply:
I am sorry, but that is clearly not correct, and that is the whole point of this pull request and the problem here.  The confusion here is that there are two different types of response extension.  That is extending another answer (really `INSERT_RESPONSE` and `append_response`), or extending a radio button (really `EXTEND_RESPONSE` and `extend_response`).  The $extend argument that comes from `BUTTONS` is meant to extend another answer.  As such it should do what I have it set up to do.  That is pass the $extend argument on to `NAMED_ANS_RADIO` that is now properly coded to extend the answer group.

If it is put back to the way that it was before then when `NAMED_ANS_RADIO_EXTENSION` is called, that method then calls `EXTEND_RESPONSE`.  That method then does nothing because `$PG->{PG_ANSWERS_HASH}->{$ans_label}` is not defined.  So no response is extended, and no answers are recorded.  Failure ensues as later calls to `NAMED_ANS_RADIO_EXTENSION` and then `EXTEND_RESPONSE` also do nothing as `$PG->{PG_ANSWERS_HASH}->{$ans_label}` is still not defined.

With the way that it is now, `NAMED_ANS_RADIO` calls `INSERT_RESPONSE` if this is an $extend case, which calls `append_response`.  That adds a response to the response group for the radio answer.  Then the later calls the `NAMED_ANS_RADIO_EXTENSION` extend that response.

Reply:
If you doubt my claims here, test this with the test problem in my initial comments for the pull request.  Uncomment the warn statement that I included.  Without this change you will see that the multianswer response has nothing for the radio answer.  While with this change there is a response added for the radio answer.

Reply:
To save you the effort of testing, here are screenshots of the resulting response group without and with this pull request respectively:

without:
![without-pr](https://user-images.githubusercontent.com/47527406/78792237-7be61700-7976-11ea-9a1e-c7cf62909458.png)

with:
![with-pr](https://user-images.githubusercontent.com/47527406/78792254-830d2500-7976-11ea-8f9d-a3befdac37ab.png)

Reply:
Sorry, I misunderstood the changes in `NAMED_ANS_RADIO()`, which now handles both initial buttons and extensions (I'm afraid I haven't followed all the changes to the answer blank infrastructure that are now in use).